src/classes/thumbnail.py

```diff
@@ -62,7 +62,7 @@ def GenerateThumbnail(file_path, thumb_path, thumbnail_frame, width, height, mas
         if reader.info.metadata.count("rotate"):
             rotate = float(reader.info.metadata.find("rotate").value()[1])
     except Exception:
-        pass
+        log.warning('Failed to parse rotation data as float')
```

Comment:
We could be even more informative, with something like,
```python

    try:
        if reader.info.metadata.has_key("rotate"):
            rotate_data = reader.info.metadata.find("rotate").value()[1]
            rotate = float(rotate_data)
    except ValueError as ex:
            log.warning("Could not parse rotation value {}: {}".format(rotate_data, ex))
    except Exception:
            log.warning("Error reading rotation metadata from {}".format(file_path), exc_info=1)
```
(That'll cover the bases, whether the exception comes from the `float()` or the previous `.find().value()[1]`)

Reply:
reader.info.metadata is not a dict, but otherwise I love it. I'll adapt this shortly.